fastlane/fastlane
# Add -packageAuthorizationProvider flag for xcodebuild

### Checklist


### Motivation and Context

With the new Xcode 16 RC, we have seen hangs in our project when resolving dependencies due to a new flag to xcodebuild that chooses the correct packageAuthorizationProvider when checking out packages.

Quoting from Apple's Xcode 16 RC release notes:
"Fixed: The new xcodebuild flag of -packageAuthorizationProvider keychain|netrc can now be used to force usage of keychain or netrc as the authorizaion provider. This improves Xcode’s package registry support. (118898849)"
[Xcode 16 RC Release Notes](https://developer.apple.com/documentation/xcode-release-notes/xcode-16-release-notes#Resolved-Issues)

This adds an option for users to choose what authorization provider (i.e. netrc / keychain). This resolves all our hang issues and has been tested locally in our product.

### Description

This PR adds a packageAuthorizationProvider option for xcodebuild to set a string with the correct authorization provider. This can be used like so:

```
run_tests(
        project: "Example.xcodeproj",
       ...
        package_authorization_provider: "netrc"   
    )
```

### Testing Steps

Please run an example app and use `package_authorization_provider: "netrc"` as an option, and make sure that when resolving dependencies, there is the `-packageAuthorizationProvider netrc` as part of the xcodebuild:
`xcodebuild -resolvePackageDependencies -scheme Foo -project Example.xcodeproj -packageAuthorizationProvider netrc`
